src/modules/utils/directory_nix.rs

```diff
@@ -1,54 +1,17 @@
-use nix::sys::stat::Mode;
-use nix::unistd::{Gid, Uid};
-use std::fs;
-use std::os::unix::fs::MetadataExt;
-use std::os::unix::fs::PermissionsExt;
+use nix::errno::Errno::ENOENT;
+use nix::unistd::access;
+use nix::unistd::AccessFlags;
 use std::path::Path;
 
 /// Checks if the current user can write to the `folder_path`.
-///
-/// It extracts Unix access rights from the directory and checks whether
-/// 1) the current user is the owner of the directory and whether it has the write access
-/// 2) the current user's primary group is the directory group owner whether if it has write access
-/// 2a) (not implemented on macOS) one of the supplementary groups of the current user is the
-/// directory group owner and whether it has write access
-/// 3) 'others' part of the access mask has the write access
-#[allow(clippy::useless_conversion)] // On some platforms it is not u32
 pub fn is_write_allowed(folder_path: &Path) -> Result<bool, String> {
-    let meta =
-        fs::metadata(folder_path).map_err(|e| format!("Unable to stat() directory: {e:?}"))?;
-    let perms = meta.permissions().mode();
-
-    let euid = Uid::effective();
-    if euid.is_root() {
-        return Ok(true);
-    }
-
-    if meta.uid() == euid.as_raw() {
-        Ok(perms & u32::from(Mode::S_IWUSR.bits()) != 0)
-    } else if (meta.gid() == Gid::effective().as_raw())
-        || (get_supplementary_groups().contains(&meta.gid()))
-    {
-        Ok(perms & u32::from(Mode::S_IWGRP.bits()) != 0)
-    } else {
-        Ok(perms & u32::from(Mode::S_IWOTH.bits()) != 0)
-    }
-}
-
-#[cfg(all(unix, not(any(target_os = "macos", target_os = "ios"))))]
-fn get_supplementary_groups() -> Vec<u32> {
-    match nix::unistd::getgroups() {
-        Err(_) => Vec::new(),
-        Ok(v) => v.into_iter().map(nix::unistd::Gid::as_raw).collect(),
+    match access(folder_path, AccessFlags::W_OK) {
```

Comment:
Where did the previous implementation have issues? Also, I feel like `eaccess` should be used instead.

Reply:
I had issues where a directory is writable but that function determined it was not writable. That directory is mounted by `sshfs`.
It is easy to switch to `eaccess` or `faccessat` with `AT_EACCESS` flag. Both of them have some unsupported `target_os` or `target_env`.
https://github.com/nix-rust/nix/blob/master/src/fcntl.rs#L26-L38
https://github.com/nix-rust/nix/blob/master/src/unistd.rs#L3418
Why do we need effective ids for this use case?

Reply:
Wouldn't switching to regular ids be a regression compared to the previous behavior? It was measuring the effective ability to write before. I would rather not keep around the old code, but I feel like `eaccess` should be used on those platforms that support it.

Reply:
`eaccess` doesn't work on `macOS`.